File: chatroom/android/app/src/main/java/com/example/chatroom/participants/BluetoothParticipant.kt
package com.example.chatroom.participants

import android.os.Handler
import android.os.Looper
import com.example.chatroom.core.Message
import com.example.chatroom.core.ParticipantType

/**
 * Bluetooth participant stub.
 * Full implementation TBD.
 */
class BluetoothParticipant(
    private val sessionId: String,
    private val deviceName: String,
    private val protocol: String, // "SPP" or "RFCOMM"
    private val onMessage: (Message) -> Unit
) {
    val type: ParticipantType = ParticipantType.BLUETOOTH
    val displayName: String = deviceName.ifBlank { "BLUETOOTH" }

    fun connect() {
        mainHandler.post {
            onMessage(Message(
                senderId = "bluetooth",
                senderType = ParticipantType.BLUETOOTH,
                senderName = displayName,
                content = "📱 BLUETOOTH 暂未实现 ($protocol)",
                isInfo = true
            ))
        }
    }

    fun sendInput(text: String) {
        // TBD
    }

    fun disconnect() {
        // TBD
    }

    companion object {
        private val mainHandler = Handler(Looper.getMainLooper())
    }
}

File: chatroom/ios/chatroom/Participants/BluetoothParticipant.swift
import Foundation

/// 蓝牙参与者（stub，iOS CoreBluetooth 实现）
final class BluetoothParticipant: Participant {

    let type: ParticipantType = .bluetooth
    let displayName: String

    var onMessage: ((Message) -> Void)?

    private let sessionId: String
    private let deviceName: String
    private let protocol_: String

    init(sessionId: String, deviceName: String, protocol_: String) {
        self.sessionId = sessionId
        self.deviceName = deviceName
        self.protocol_ = protocol_
        self.displayName = deviceName.isEmpty ? "BLUETOOTH" : deviceName
    }

    func connect() {
        let msg = Message(
            senderId: "bluetooth",
            senderType: .bluetooth,
            senderName: displayName,
            content: "📱 BLUETOOTH 暂未实现 (\(protocol_))",
            isInfo: true
        )
        DispatchQueue.main.async {
            self.onMessage?(msg)
        }
    }

    func sendInput(_ text: String) {
        // TBD
    }

    func disconnect() {
        // TBD
    }
}
